### PR TITLE
feat: Select preset for minimal page

### PR DESCRIPTION
**Pull request by hannahpurcell**
Asana Ticket: https://app.asana.com/0/1205385723132845/1207162343766764

### CODE REVIEW

File: lib/skate_web/router.ex
Comment (firestack) [inline lib/skate_web/router.ex:109]: question(non-blocking): what does the `*` mean here? I'm used to something like `:id` but not `*id`, is this a way to get phoenix to ignore the parameter but know it's there?

File: assets/src/components/minimalLadder.tsx
Comment (firestack) [inline assets/src/components/minimalLadder.tsx:14]: Does typescript complain if we do it like this?

```suggestion
  const { id } = useParams()
```

> https://reactrouter.com/en/main/hooks/use-params

**Review by firestack — COMMENTED** [review]
praise: loving this so far! Not able to finish review today but I've left a few comments.

File: assets/src/components/app.tsx
Comment (joshlarson) [inline assets/src/components/app.tsx:77]: Question: At this point, do we need both of these?

File: assets/src/components/minimalLadder.tsx
Comment (joshlarson) [reply assets/src/components/minimalLadder.tsx:14]: It would type `id` as a `string | undefined`, which then means that it wouldn't be passable into `routeTabById`.

This is an interesting one, because `id` _should_ always be there because of how our `ReactRouter` is configured, but I wonder how to convince Typescript of that fact.

In theory, 👇 would work, but I'm not sure I love it, given that it's more verbose
```typescript
const { id } = useParams() as { id: string }
```

Comment (firestack) [reply assets/src/components/minimalLadder.tsx:14]: Or re-stated, create a `<MinimalLadder.FromRouterParam>` which wraps `<MinimalLadder>` and does all the `useParams` stuff there (like asserting the id is specified or showing an error message or redirecting otherwise) and then calls (condensed, since you _may_ know via typescript that `id` is defined at this point.)
```
{id && <MinimalLadder id={id} />}
```

Comment (joshlarson) [reply assets/src/components/minimalLadder.tsx:14]: Yeah that makes sense to me!

File: lib/skate_web/router.ex
Comment (hannahpurcell) [reply lib/skate_web/router.ex:109]: Ohh huh! I didn't realize I was using both in different situations without a lot of nuance. It is more accurate to use `:` here. The `*` allows the rest of the matching url to be collected into the `id` param in the form `%{"id" => ["list", "of", "separated", "params"]}`. [Doc](https://hexdocs.pm/phoenix/Phoenix.Router.html#module-routing)

File: assets/src/components/minimalLadder.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadder.tsx:14]: Ha! I'm not sure why I took the awkward route here. Yes, that's great

**Comment by joshlarson** [comment]
This makes a lot of sense to me! Consider me approving once Kayla approves.

I have some thoughts and questions about UX, but these are thoughts that should _maybe_ be brought up to design, and make the most sense as follow-ups:
- I still don't think that the minimal view should have any of the interactive components. I can add a follow-up task to get rid of them if you all agree.
- Right now it's not particularly discoverable. Should there be a way to navigate to `/minimal`? Like a new nav link in the nav bar gated behind the same test group? I'm torn about this because it raises some thorny questions about users/roles/flags/etc (basically this would be perpetually available only to one user), but still.
- Should there be some kind of error state in case the preset that you tried to refresh gets deleted? (Or in case you navigate to `/minimal/oops` or something like that)

File: assets/src/components/minimalLadderPage.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadderPage.tsx:22]: Yes, I can do this with some variation: the reactstrap Button doesn't like the prop `as`, so instead I can wrap the Button in a NavLink.

If I use the navless Button, the UX is that the page is blank for a moment before the route ladders render fully loaded. If I use the NavLink approach, the page renders kinda wonky before filling out. What is the benefit to going with NavLink here?

May-13-2024 17-21-44

File: assets/src/components/minimalLadder.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadder.tsx:14]: Omg once again, I didn't refresh to get all the comments. Ignore my last one, I'm re-reading

**Comment by hannahpurcell** [comment]
@joshlarson agree 1 and 2 are questions for both design & product lead. While the polish would be nice, I'd guess that once this "minimal page" is shipped, we have other detour priorities to work on before finessing this page

File: assets/src/components/minimalLadder.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadder.tsx:14]: Looking at this with fresh eyes. While creating the `.FromRouterParam` may be the proper way, I don't naturally prefer this approach. I'll do it if I must! But it adds a layer of, like, unneeded padding around the MinimalLadder component that I think won't actually come in handy down the road? It's a layer of extra stuff around MinimalLadder that will add complexity in the App component, to fix one line in MinimalLadder that could be a little tighter. I suppose I find the existing one line more readable. Is that a bad take?

File: assets/src/components/minimalLadderPage.tsx
Comment (firestack) [reply assets/src/components/minimalLadderPage.tsx:22]: 1. I realize we don't _need_ `NavLink` specifically, but I'd like to use the [`Link` from `reactrouter`](https://reactrouter.com/en/main/components/link) since this is a url that is managed by reactrouter.
2. Is the difference in UX shown in the video posted above? I'm having a hard time seeing the difference

File: assets/src/components/minimalLadder.tsx
Comment (firestack) [inline assets/src/components/minimalLadder.tsx:17]: suggestion: can we use [`useNavigate`](https://reactrouter.com/en/main/hooks/use-navigate)?

File: assets/src/components/minimalLadderPage.tsx
Comment (firestack) [inline assets/src/components/minimalLadderPage.tsx:23]: suggestion: Can we use bootstrap to style and lay these out?
```suggestion
      <Nav>
        {presets.map((preset) => (
          <Nav.Item key={preset.uuid}>
            <Nav.Link as={Link} to={"/minimal/" + preset.uuid}>
              {preset.presetName}
            </Nav.Link>
          </Nav.Item>
        ))}
      </Nav>
```

File: assets/src/components/minimalLadder.tsx
Comment (firestack) [inline assets/src/components/minimalLadder.tsx:24]: nit(non-blocking)
```suggestion
  const { selectedRouteIds = [], ladderDirections = {}, ladderCrowdingToggles = {} } =
    routeTab || {}
```

File: assets/src/components/minimalLadder.tsx
Comment (firestack) [reply assets/src/components/minimalLadder.tsx:14]: I'm not okay with the type coersion (the `as string`), as I feel that's somewhat defeating the usefulnes of typescript, so I think you should handle the `undefined` case explicitly, but however you want to handle that is fine (if you don't feel like making a wrapper component to handle that logic)

Comment (firestack) [reply assets/src/components/minimalLadder.tsx:14]: In terms of reability, I don't find it harder to read, and I find it easier to reason about when the required parameter is just a parameter declared the correct type, but I this this is really just style and taste and nothing to worry about any specific way of doing it.

File: assets/css/app.scss
Comment (firestack) [inline assets/css/app.scss:73]: note: I don't think we'll need this if you do https://github.com/mbta/skate/pull/2588#discussion_r1600363159

File: assets/src/components/minimalLadderPage.tsx
Comment (firestack) [reply assets/src/components/minimalLadderPage.tsx:22]: Ended up sketching out an idea https://github.com/mbta/skate/pull/2588#discussion_r1600363159

Comment (firestack) [reply assets/src/components/minimalLadderPage.tsx:22]: Also I tried the UX of a regular link vs a react router link, and I feel that the loading is smoother because it's a SPA transition when using `<Link>`, but it's a regular HTTP request when using `<Button href={} />` which ends up doing full page layout and first time rendering again.

File: assets/src/components/minimalLadder.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadder.tsx:14]: Ohhhkay finally implemented, and I understand it better! Thanks for your patience 🙏

File: assets/src/components/minimalLadderPage.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadderPage.tsx:22]: Ahhhh ok. I see why the react router link choppiness is still a positive tradeoff, then. Thanks! 👍

File: assets/src/components/minimalLadderPage.tsx
Comment (hannahpurcell) [reply assets/src/components/minimalLadderPage.tsx:23]: Hmmmm I've waffled a bit. My default inclination was to use buttons, and I'm not married to that. But they _are_ bootstrap, and I aesthetically like them. It feels nitpicky to not just commit the above, but since I have to add the imports anyway, I think I'm going to commit a blend that I have, as noted and seen in [this comment](https://github.com/mbta/skate/pull/2588#discussion_r1599100828), but I'll keep the Link

File: assets/css/_minimal_ladder_page.scss
Comment (hannahpurcell) [reply assets/css/_minimal_ladder_page.scss:3]: It's used in MinimalLadderPage, as container padding

File: assets/tests/components/__snapshots__/minimalLadderPage.test.tsx.snap
Comment (firestack) [inline assets/tests/components/__snapshots__/minimalLadderPage.test.tsx.snap:21]: issue: Acording to the anchor element's "Permitted content":

> Permitted content:
>> Transparent, except that no descendant may be interactive content or an <a> element, and no descendant may have a specified tabindex attribute. 
>
> -- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#technical_summary

I believe a `button` is ["interactive content"](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#interactive_content) and I think we're really only using this for styling?

What if you put `btn btn-primary` on the `<a>`?

File: assets/css/_minimal_ladder_page.scss
Comment (firestack) [reply assets/css/_minimal_ladder_page.scss:3]: Thanks! Sorry I missed it!

File: assets/src/components/minimalLadderPage.tsx
Comment (firestack) [reply assets/src/components/minimalLadderPage.tsx]: I think this is related? https://github.com/mbta/skate/pull/2588#discussion_r1603608483

**Review by firestack — APPROVED** [review]

**Pull request merged**
